github/copilot.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// CopilotService provides access to the Copilot-related functions
+// in the GitHub API.
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/
+type CopilotService service
+
+// OrganizationCopilotDetails represents the details of an organization's Copilot for Business supbscription.
+type OrganizationCopilotDetails struct {
+	SeatBreakdown         *SeatBreakdown `json:"seat_breakdown"`
+	PublicCodeSuggestions string         `json:"public_code_suggestions"`
+	CopilotChat           string         `json:"copilot_chat"`
+	SeatManagementSetting string         `json:"seat_management_setting"`
+}
+
+// SeatBreakdown represents the breakdown of Copilot for Business seats for the organization.
+type SeatBreakdown struct {
+	Total               int64 `json:"total"`
+	AddedThisCycle      int64 `json:"added_this_cycle"`
+	PendingCancellation int64 `json:"pending_cancellation"`
+	PendingInvitation   int64 `json:"pending_invitation"`
+	ActiveThisCycle     int64 `json:"active_this_cycle"`
+	InactiveThisCycle   int64 `json:"inactive_this_cycle"`
+}
+
+type CopilotSeats struct {
+	TotalSeats int64                `json:"total_seats"`
+	Seats      []CopilotSeatDetails `json:"seats"`
+}
+
+// CopilotSeatDetails represents the details of a Copilot for Business seat.
+// Assignee can either be a User, Team, or Organization.
+type CopilotSeatDetails struct {
+	Assignee                interface{} `json:"assignee"`
+	AssigningTeam           *Team       `json:"assigning_team,omitempty"`
+	PendingCancellationDate *string     `json:"pending_cancellation_date,omitempty"`
+	LastActivityAt          *string     `json:"last_activity_at,omitempty"`
+	LastActivityEditor      *string     `json:"last_activity_editor,omitempty"`
+	CreatedAt               string      `json:"created_at"`
+	UpdatedAt               string      `json:"updated_at,omitempty"`
+}
+
+// SelectedTeams represents the teams selected for the Copilot for Business subscription.
+type SelectedTeams struct {
+	SelectedTeams []string `json:"selected_teams"`
+}
+
+// SelectedUsers represents the users selected for the Copilot for Business subscription.
+type SelectedUsers struct {
+	SelectedUsers []string `json:"selected_users"`
+}
+
+// SeatAssignments represents the number of seats assigned.
+type SeatAssignments struct {
+	SeatsCreated int64 `json:"seats_created"`
```

Comment:
I think this can safely be an `int`. We typically reserve `int64` for IDs, and counts we usually use `int`.

```suggestion
	SeatsCreated int `json:"seats_created"`
```